spec/factories/miq_user_role.rb

```diff
@@ -1,11 +1,22 @@
 FactoryGirl.define do
+  sequence(:miq_user_role_name) { |n| "UserRole #{seq_padded_for_sorting(n)}" }
+
   factory :miq_user_role do
-    name "Test Role"
-  end
+    transient do
+      # e.g.: miq_request_approval
+      features nil
+      # e.g.: super_administrator
+      evm      nil
+    end
 
-  factory :miq_user_role_miq_request_approver, :parent => :miq_user_role do
-    sequence(:name) { |n| "Request Approver #{seq_padded_for_sorting(n)}" }
+    name { |ur| ur.evm ? "EvmRole-#{ur.evm}" : generate(:miq_user_role_name) }
 
-    miq_product_features { [FactoryGirl.create(:miq_product_feature_miq_request_approval)] }
+    after(:build) do |user, evaluator|
+      if evaluator.features.present?
+        user.miq_product_features = Array.wrap(evaluator.features).map do |f|
+          f.kind_of?(MiqProductFeature) ? f : FactoryGirl.create(:miq_product_feature, :identifier => f)
```

Comment:
Should that FactoryGril.create be wrapped in an Array?  Otherwise won't it set a single product feature to an association?